Basics of reading ASDF files.ipynb

```diff
@@ -0,0 +1,1084 @@
+      "\u001b[2m│ ├─\u001b[0m\u001b[1mauthor\u001b[0m (str): Space Telescope Science Institute\n",
+      "\u001b[2m│ ├─\u001b[0m\u001b[1mhomepage\u001b[0m (str): http://github.com/spacetelescope/asdf\n",
+      "\u001b[2m│ ├─\u001b[0m\u001b[1mname\u001b[0m (str): asdf\n",
+      "\u001b[2m│ └─\u001b[0m\u001b[1mversion\u001b[0m (str): 2.7.1\n",
+      "\u001b[2m├─\u001b[0m\u001b[1mhistory\u001b[0m (dict)\n",
+      "\u001b[2m│ └─\u001b[0m\u001b[1mextensions\u001b[0m (list)\u001b[3m ...\u001b[0m\n",
+      "\u001b[2m├─\u001b[0m\u001b[1m_fits_hash\u001b[0m (str): 18bd2f439e9aa9d5cbc184c5518799b7978d665fdee1db550d4dadc4b3e3028d\n",
+      "\u001b[2m├─\u001b[0m\u001b[1marea\u001b[0m (NDArrayType): shape=(2048, 2048), dtype=float32\n",
+      "\u001b[2m├─\u001b[0m\u001b[1mdata\u001b[0m (NDArrayType): shape=(2048, 2048), dtype=float32\n",
+      "\u001b[2m├─\u001b[0m\u001b[1mdq\u001b[0m (NDArrayType): shape=(2048, 2048), dtype=uint32\n",
+      "\u001b[2m├─\u001b[0m\u001b[1merr\u001b[0m (NDArrayType): shape=(2048, 2048), dtype=float32\n",
+      "\u001b[2m├─\u001b[0m\u001b[1mextra_fits\u001b[0m (dict)\n",
+      "\u001b[2m│ └─\u001b[0m\u001b[1mPRIMARY\u001b[0m (dict)\u001b[3m ...\u001b[0m\n",
+      "\u001b[2m├─\u001b[0m\u001b[1mmeta\u001b[0m (dict)\n",
+      "\u001b[2m│ ├─\u001b[0m\u001b[1masn\u001b[0m (dict)\u001b[3m ...\u001b[0m\n",
+      "\u001b[2m│ ├─\u001b[0m\u001b[1mbunit_data\u001b[0m (str): MJy/sr\n",
+      "\u001b[2m│ ├─\u001b[0m\u001b[1mbunit_err\u001b[0m (str): MJy/sr\n",
+      "\u001b[2m│ └─\u001b[0m\u001b[2m\u001b[3m24 not shown\u001b[0m\u001b[0m\n",
+      "\u001b[2m├─\u001b[0m\u001b[1mvar_flat\u001b[0m (NDArrayType): shape=(2048, 2048), dtype=float32\n",
+      "\u001b[2m├─\u001b[0m\u001b[1mvar_poisson\u001b[0m (NDArrayType): shape=(2048, 2048), dtype=float32\n",
+      "\u001b[2m└─\u001b[0m\u001b[1mvar_rnoise\u001b[0m (NDArrayType): shape=(2048, 2048), dtype=float32\n",
+      "\u001b[2m\u001b[3mSome nodes not shown.\u001b[0m\u001b[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "af.info()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The info() method is extremely useful for examining the contents of an ASDF file. There is a command line version as well discussed later. But first we'll do a little data access. Note all the items labeled as NDArrayTypes. As you might guess, these are all images so we will access the main science image, which has te name \"data\". First thing to know is that when you open ASDF files, images are not loaded into memory when it is opened, but information about that image (or array) is. When you actually access the data itself, it will be loaded. We will do that now"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Look at the science data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 114,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
```

Comment:
It would be good to time to show that names aren't limited like FITS...can we make the file contain "science_data" instead of "data" etc..."data_quality" instead of "dq"....

Reply:
Separate notebooks for general asdf and jwst-asdf and roman-asdf would be instructive